trip_offer_service/src/offer/domain/queries/get_offer_query.py

```diff
@@ -0,0 +1,21 @@
+from uuid import UUID
+
+from flask import Config
+
+from src.infrastructure.storage import MongoReadOnlyClient
+from src.offer.domain.ports import IGetOfferQuery
+from src.offer.infrastructure.storage.offer import Offer
+
+
+class GetOfferQuery(IGetOfferQuery):
+    def __init__(self, config: Config, client: MongoReadOnlyClient) -> None:
+        self.collection_name = config["MONGO_VIEW_COLLECTION_NAME"]
+        self.client = client
+
+    def get_offer(self, offer_id: UUID) -> Offer:
+        result = self.client.get_db()[self.collection_name].find_one(
+            {"offer_id": str(offer_id)}
+        )
+
+        assert result is not None
```

Comment:
tutaj nie możemy zrobić takiego asserta, jeżeli nie ma takiego rekordu w bazie to powinniśmy zwrócić stosowną informację
```suggestion
    def get_offer(self, offer_id: UUID) -> Offer:
        result = self.client.get_db()[self.collection_name].find_one(
            {"offer_id": str(offer_id)}
        )
        if not result:
            raise OfferNotFoundException
```
I exceptiona obsłużyć w `Resource'ie` i zwróci informację do użytkonika 'offer_not_found_error'